app/models/pq.rb

```diff
@@ -84,6 +83,72 @@ def self.ministers_by_progress(ministers, progresses)
       count
   end
 
+  def find_or_create_follow_up
+    unless i_will_write && follow_up_to.blank? && valid?
+      raise "A PQ follow-up must be valid, have 'i_will_write' set to true, and 'follow_up_to' blank!"
+    end
+
+    ActiveRecord::Base.transaction do
+      follow_up = Pq.find_or_initialize_by(uin: iww_uin)
+
+      if follow_up.new_record?
+        attrs = attributes
+                  .reject { |k,_| FOLLOW_UP_ATTR_RESET.include?(k) }
+                  .merge('question_type' => 'Follow-up IWW',
+                         'i_will_write'  => true,
+                         'follow_up_to'  => uin,
+                         'progress'      => Progress.draft_pending)
+
+        follow_up.update!(attrs)
+      end
+
+      ao_pq = action_officers_pqs.find(&:accepted?)
```

Comment:
Shouldn't we create a new action_officers_pq record, instead of updating the old one? Check with Rose